internal/pkg/agent/configuration/fleet_server.go

```diff
@@ -14,6 +15,7 @@ import (
 // FleetServerConfig is the configuration written so Elastic Agent can run Fleet Server.
 type FleetServerConfig struct {
 	Bootstrap    bool                     `config:"bootstrap" yaml:"bootstrap,omitempty"`
+	InitTimeout  *time.Duration           `config:"fleet_timeout" yaml:"fleet_timeout"`
```

Comment:
Your already under `fleet.server` so the `fleet_timeout` with the `fleet_` prefix seems redundant and weird. I would go with `init_timeout`.